install_files/ansible-base/roles/tails-config/files/securedrop@securedrop.freedom.press/extension.js

```diff
@@ -0,0 +1,129 @@
+
+        // Check for SSH keys - if they don't exist, this is a Journalist workstation
+        // (and therefore the admin options should remain hidden)
+        let is_admin = false;
+        let [ok, out, err, exit] = GLib.spawn_command_line_sync('/bin/bash -c "test -f ~/Persistent/securedrop/install_files/ansible-base/app-ssh.auth_private"');
+        if (exit == 0) {
+            is_admin = true;
+        }
+
+        let icon = new St.Icon({
+            icon_name: 'package-x-generic-symbolic',
+            style_class: 'system-status-icon',
+        });
+        
+        let label = new St.Label({
+            text: 'SecureDrop',
+            y_expand: true,
+            y_align: Clutter.ActorAlign.CENTER,
+        });
+        
+        let topBox = new St.BoxLayout({
+            style_class: 'panel-status-menu-box'
+        });
+        
+        topBox.add_child(icon);
+        topBox.add_child(label);
+        this.add_child(topBox);
+
+        let source = new PopupMenu.PopupMenuItem('Launch Source Interface');
+        source.connect('activate', () => {
+            let [ok, out, err, exit] = GLib.spawn_command_line_sync('/bin/bash -c "cat ~/Persistent/securedrop/install_files/ansible-base/app-sourcev3-ths"');
+       	    if (out.length > 0) {
+       	    	Util.spawn(['tor-browser', out.toString()]);
+       	    } else {
+       	    	Util.trySpawnCommandLine('notify-send "Unable to find Source Interface Onion Address"');
+       	    }
+        });
+        
+        let journalist = new PopupMenu.PopupMenuItem('Launch Journalist Interface');
+        journalist.connect('activate', () => {
+       	    let [ok, out, err, exit] = GLib.spawn_command_line_sync('/bin/bash -c "cut -d: -f1 ~/Persistent/securedrop/install_files/ansible-base/app-journalist.auth_private"');
+       	    if (out.length > 0) {
+       	    	Util.spawn(['tor-browser', out.toString() + '.onion']);
+       	    } else {
+       	    	Util.trySpawnCommandLine('notify-send "Unable to find Journalist Interface Onion Address"');
+       	    }
+        });
+        
+        let admin_label = new St.Label({
+            text: 'Admin Actions',
+            y_expand: true,
+            y_align: Clutter.ActorAlign.START,
+        });
+        
+        let updates = new PopupMenu.PopupMenuItem('Check for SecureDrop Updates');
+        updates.connect('activate', () => {
+            Util.spawn(['python3', '/home/amnesia/Persistent/securedrop/journalist_gui/SecureDropUpdater']);
+        });
+        
+        let app_server_ssh = new PopupMenu.PopupMenuItem('SSH into the App Server');
+        app_server_ssh.connect('activate', () => {
+            Util.trySpawnCommandLine(`gnome-terminal -- ssh app`);
+        });
+        
+        let mon_server_ssh = new PopupMenu.PopupMenuItem('SSH into the Monitor Server');
+        mon_server_ssh.connect('activate', () => {
+            Util.trySpawnCommandLine(`gnome-terminal -- ssh mon`);
```

Comment:
It's not guaranteed that the server alias is 'mon', the value set in sdconfig is used in the .ssh/config file.